physics/GFS_surface_composites_post.F90

```diff
@@ -260,11 +266,38 @@ subroutine GFS_surface_composites_post_run (
           endif
         enddo
 
-      else
+      else ! not fractional grid
 
         do i=1,im
-          if (islmsk(i) == 1) then
+
+          if (use_lake_model(i)>0) then
+            if(frac_ice .and. icy(i)) then
+              call composite_icy(iopt_lake==iopt_lake_clm)
+              call composite_wet_and_icy
+            else
+              call composite_wet
+            endif
+          else if (islmsk(i) == 1) then
           !-- land
+            call composite_land
+          elseif (islmsk(i) == 0) then
+          !-- water
+            call composite_wet
+          else ! islmsk(i) == 2
+          !-- ice
+            call composite_icy(.false.)
+            call composite_wet_and_icy
```

Comment:
@SamuelTrahanNOAA Do we need to have  on line 289:
if (frac_ice) call composite_wet_and_icy 
or
if(cice(i) < 1)  call composite_wet_and_icy

Reply:
I haven't tested fractional ice on ocean points, so I don't know if that would work. It is set up to only use fractional ice on lake points.